pycds/alembic/versions/7b139906ac46_add_not_null_constraint_on_meta_vars.py

```diff
@@ -0,0 +1,85 @@
+"""add not null constraint on meta vars
+
+Revision ID: 7b139906ac46
+Revises: 3d50ec832e47
+Create Date: 2022-01-17 14:21:37.975335
+
+"""
+from alembic import op
+import sqlalchemy as sa
+from pycds import get_schema_name
+
+# revision identifiers, used by Alembic.
+revision = "7b139906ac46"
+down_revision = "3d50ec832e47"
+branch_labels = None
+depends_on = None
+
+schema_name = get_schema_name()
+table_name = "meta_vars"
+
+
+def update_table(
+    var,
+    replacement,
+):
+    op.execute(
+        f"""UPDATE {schema_name}.{table_name} SET {var}={replacement} WHERE {var} IS null;"""
+    )
+
+
+def regress_table(
+    var,
+    replacable,
+):
+    op.execute(
+        f"""UPDATE {schema_name}.{table_name} SET {var}=NULL WHERE {var} LIKE {replacable};"""
+    )
```

Comment:
Interesting. Downgrades should be, whenever possible, full inverses of upgrades, and this accomplishes that even for the data migration (assuming database content is not changed before downgrading). It's debatable whether that's actually necessary in this particular case, but I like your attention to detail.